File: factorial/fact.py
def factorial(n):
    if n == 0 or n == 1:
        return 1
    return n * factorial(n - 1)


try:
    user_input = int(input("Enter a integer: "))
    
    if user_input < 0:
        print("Please enter a non-negative integer.")
    else:
        result = factorial(user_input)
        print(f"The factorial of {user_input} is {result}.")
except ValueError:
    print("Invalid input! Please enter a valid non-negative integer.")

File: recursive/fact.js
function factorial(n) {
    if (n < 0) {
        throw new Error("Factorial is not defined for negative numbers");
    }
    if (n === 0 || n === 1) {
        return 1;
    }
    return n * factorial(n - 1);
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Enter a number: ', number => {
    const n = parseInt(number);
    if (isNaN(n)) {
        console.log("Please enter a valid number.");
    } else {
        try {
            console.log(`Factorial of ${n} is ${factorial(n)}`);
        } catch (error) {
            console.log(error.message);
        }
    }
    readline.close();
});
